### PR TITLE
Notification mail for new registrations

### PR DESCRIPTION
**Pull request by machiato32**
Emails go out to the admins and the secretaries.
Added new mail object, blade and a few translations.

### CODE REVIEW

File: app/Http/Controllers/Auth/RegisterController.php [hunk 1]
```diff
@@ -172,6 +173,15 @@ protected function create(array $data)
...
         // Send confirmation mail.
         Mail::to($user)->queue(new \App\Mail\Confirmation($user->name));
+        // Send notification about new user to the admins and to the secretaries.
+        $admins = User::role(Role::NETWORK_ADMIN)->get();
+        $secretaries = User::role(Role::SECRETARY)->get();
+        foreach ($admins as $admin) {
+            Mail::to($admin)->send(new NewRegistration($admin->name, $user->name, $user->isCollegist()));
+        }
+        foreach ($secretaries as $secretary) {
+            Mail::to($secretary)->send(new NewRegistration($secretary->name, $user->name, $user->isCollegist()));
```
Comment (kdmnk) [inline app/Http/Controllers/Auth/RegisterController.php:183]: Here passing only $user is enough and then you can handle this in the constructor.

File: resources/lang/hu/mail.php [hunk 2]
```diff
@@ -5,6 +5,9 @@
     'approved_registration' => 'A regisztrációdat jóváhagyták.',
     'dear' => 'Kedves',
     'email_sent' => 'Rendszergazdák értesítve',
+    'new-registration-subject' => 'Új regisztráció',
+    'new_registration_collegist' => ':registered_user collégistaként regisztrált az Urán rendszerébe. Fogadd el vagy utasítsd vissza a kérelmét.',
+    'new_registration_tenant' => ':registered_user vendégként regisztrált az Urán rendszerébe. Fogadd el vagy utasítsd vissza a kérelmét.',
```
Comment (kdmnk) [inline resources/lang/hu/mail.php:10]: Merge these. Collegist/Tenant status can be shown separately.

File: resources/views/emails/new_registration.blade.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+@component('mail::message')
+    <h1>@lang('mail.dear') {{ $recipient }}!</h1>
+    <p>
+        @if ($is_collegist)
+            @lang('mail.new_registration_collegist', ['registered_user' => $registered_user])
+        @else
+            @lang('mail.new_registration_tenant', ['registered_user' => $registered_user])
+        @endif
+    </p>
```
Comment (kdmnk) [inline resources/views/emails/new_registration.blade.php:9]: Add some basic data (status, email, etc.) and a button with a link the corresponding page.

File: app/Mail/NewRegistration.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Create a new message instance.
+     *
+     * @return void
+     */
+    public function __construct(string $recipient, string $registered_user, bool $is_collegist)
+    {
+        $this->recipient = $recipient;
+        $this->registered_user = $registered_user;
+        $this->is_collegist = $is_collegist;
```
Comment (kdmnk) [inline app/Mail/NewRegistration.php:26]: Here passing only $user is enough. This way you can use the user's other attributes later as well.

**Review by kdmnk — CHANGES_REQUESTED** [review]

File: resources/views/emails/new_registration.blade.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+@lang('user.place_and_date_of_birth'): {{$personal->getPlaceAndDateOfBirth()}}<br>
+@lang('user.mothers_name'): {{$personal->mothers_name}}<br>
+@lang('user.country'): {{$personal->country}}<br>
+@lang('user.tenant_until'): {{$personal->tenant_until}}
+@endcomponent
+<div class="row">
+@component('mail::button', ['url'=> route('secretariat.registrations.accept', ['id' => $user->id])])<!--todo url-->
+@lang('user.accept')    
+@endcomponent
+@component('mail::button', ['url'=> route('secretariat.registrations.show', ['id' => $user->id])])<!--todo url-->
```
Comment (kdmnk) [inline resources/views/emails/new_registration.blade.php:20]: remove todos

**Review by kdmnk — CHANGES_REQUESTED** [review]
The Staff do not have permission to handle registrations.
You can find this as a Gate in AuthServiceProvider, for now I think its enough to add the Staff to it. @luksan47 will restructure this anyway. (Why are we using Gates instead of policies?)

**Comment by machiato32** [comment]
Secretaries only see the new registrations of collegists, the staff can handle the teanants' registrations.

**Comment by machiato32** [comment]
Accepting and rejecting registrations should only be possible by users with permissions to do so (staff->tenant, secretaries->collegist)

**Comment by kdmnk** [comment]
@machiato32 
The english hungarian consistency test fails (it might be only a bad ordering, try php artisan locale:sort).


> Accepting and rejecting registrations should only be possible by users with permissions to do so (staff->tenant, secretaries->collegist)

Create a new issue for this, we'll merge this and release.

**Comment by kdmnk** [comment]
@luksan47 since I cannot test right now, can you test if the staff is able to approve the guests today?
Or we can just release and possibly add a hotfix :D

**Review by kdmnk — APPROVED** [review]

**Pull request merged**